### ISSUE TITLE
Yul compilation failed: Call or assignment expected

### ISSUE THREAD

**Issue by agroce** [ethereum/fe#293]
Compiling this file:

```
contract FooProxy:
    pub def baz():
        ()
```

```
thread 'main' panicked at 'Yul compilation failed: {"errors":[{"component":"general","formattedMessage":"ParserError: Call or assignment expected.\n --> input.yul:1:168:\n  |\n1 |  ... e\" { code { function $$baz() { 0x0 } function avail() -> ptr { ptr := m ...\n  |                                         ^\n\n","message":"Call or assignment expected.","severity":"error","sourceLocation":{"end":168,"file":"input.yul","start":167},"type":"ParserError"},{"component":"general","formattedMessage":"ParserError: Expected keyword \"data\" or \"object\" or \"}\".\n --> input.yul:1:170:\n  |\n1 |  ...  { code { function $$baz() { 0x0 } function avail() -> ptr { ptr := mload(0x00 ...\n  |                                         ^^^^^^^^\n\n","message":"Expected keyword \"data\" or \"object\" or \"}\".","severity":"error","sourceLocation":{"end":177,"file":"input.yul","start":169},"type":"ParserError"}]}', compiler/src/lib.rs:48:27
stack backtrace:
   0: rust_begin_unwind
             at /rustc/9c09c1f7cfcf9de0522bcd1cfda32b552195c464/library/std/src/panicking.rs:493:5
   1: std::panicking::begin_panic_fmt
             at /rustc/9c09c1f7cfcf9de0522bcd1cfda32b552195c464/library/std/src/panicking.rs:435:5
   2: fe_compiler::compile
   3: fe::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
Aborted
```

On latest github master, built with

```
[profile.dev]
debug = true
panic = "abort"
```

for fuzzing.  Using https://github.com/agroce/afl-compiler-fuzzer

**Comment by cburgdorf** [comment]
This will be resolved with #187 (tuple support). It's probably not worth fixing now because after lowering, Tuples won't be part of the AST that comes out of lowering (they'll be converted to structs at that point.)

**Comment by sbillig** [comment]
This issue still persists, even with the recent work on tuples.
